skyportal/handlers/api/comment.py

```diff
@@ -87,15 +87,56 @@ def post(self):
         obj_id = data['obj_id']
         # Ensure user/token has access to parent source
         _ = Source.get_obj_if_owned_by(obj_id, self.current_user)
-        user_group_ids = [g.id for g in self.current_user.groups]
         user_accessible_group_ids = [g.id for g in self.current_user.accessible_groups]
-        group_ids = data.pop("group_ids", user_group_ids)
-        group_ids = [gid for gid in group_ids if gid in user_accessible_group_ids]
+        user_accessible_filter_ids = [
+            filtr.id
+            for g in self.current_user.accessible_groups
+            for filtr in g.filters
+            if g.filters is not None
+        ]
+        group_ids = data.pop("group_ids", user_accessible_group_ids)
```

Comment:
Can this be more succinctly written as an intersection operation?

Reply:
the `data.pop` statement? I'm not sure what you mean... `user_accessible-groups` is the default value passed to `pop`

Reply:
Sorry, this comment should have been one line up.